src/main/java/io/anserini/collection/C4Collection.java

```diff
@@ -80,8 +80,8 @@ private int getFileNumber(String fileName) {
   public static class Segment extends FileSegment<C4Collection.Document>{
     private MappingIterator<JsonNode> iterator; // iterator for JSON line objects
     private JsonNode node = null;
-    private String filePath;
-    private String fileName;
+    private static String filePath;
```

Comment:
I don't think you want static here? `protected`?

Reply:
Yes, but I didn't want to change attribute visibility of previously developed code without first consulting with you. I have already changed them :)